### DISCUSSION TITLE
Option to empty cache for main cache plugins after a post expires

### DISCUSSION THREAD

**Discussion by nicomollet** [publishpress/publishpress-future#349] [Feature Requests]
Hello,

I think PublishePress Future should handle cache plugins.

**Is your feature request related to a problem? Please describe.**
Expired posts might still be online because cache plugins don't refresh the contents.

**Describe the solution you'd like**
I would like an option "Empty cache after a post expires". This option would trigger the "empty cache" functions of the main cache plugins (WP Super Cache, WP Rocket, etc).

**Describe alternatives you've considered**
I hooked into the hook `postexpiratior_unschedule` to trigger the empty cache function of my cache plugin (WP Rocket).

Thank you

**Comment by agapetry** [comment]
Are we talking about a full cache flush with wp_cache_flush() or a selective flush with wp_cache_delete() ?

**Comment by nicomollet** [comment]
It's not related to object cache.

This is about plugin territory, cache plugins that create static page cache. 
The expired posts (by your plugin) are still be accessible if the page cache is not emptied.

**Comment by andergmartins** [comment]
Regarding being a full cache refresh, or selective refreshing, that would depend on the available options for each cache plugin we support. But the idea makes sense to me. It could be an option setting, deactivated by default.

**Comment by illycz** [comment]
@nicomollet Can you post example about "I hooked into the hook postexpiratior_unschedule to trigger the empty cache function of my cache plugin (WP Rocket)." please.

**Comment by nicomollet** [comment]
@illycz something like:

```php
/**
* Post-Expirator: clear WP-Rocket cache after expiration
*/
function postexpirator_expireclearcache()
{
// Clear the cache.
  if (function_exists('rocket_clean_domain')) {
	  rocket_clean_domain();
  }
}
add_action( 'postExpiratorExpire', 'postexpirator_expireclearcache', 10 );

```
